Support XML API branch with conflicts resolved

LarsAlmgren opened the pull request:
All work done by @tustvold but since https://github.com/fsouza/fake-gcs-server/pull/1164 is stale I've resolved the conflicts and pushed a new branch to get this going.

fsouza commented on fakestorage/upload_test.go:
```diff
@@ -607,23 +603,6 @@ func TestServerClientSignedUploadBucketCNAME(t *testing.T) {
-	if obj.Md5Hash != expectedHash {
-		t.Errorf("wrong md5 hash\nwant %q\ngot  %q", expectedHash, obj.Md5Hash)
-	}
 }
```
why is this getting removed?

fsouza commented on fakestorage/upload_test.go:
```diff
@@ -738,12 +717,73 @@ func TestServerClientDeleteResumableUpload(t *testing.T) {
+		t.Errorf("got %d expected %d", resp.StatusCode, http.StatusOK)
+	}
+
+	req, err = http.NewRequest("PUT", server.URL()+"/bucket2/path", nil)
```
same as above
```suggestion
	req, _ = http.NewRequest("PUT", server.URL()+"/bucket2/path", nil)
```

fsouza reviewed (commented):
thanks for picking this up and apologies for the delayed review

LarsAlmgren commented:
> thanks for picking this up and apologies for the delayed review


@fsouza No worries, it goes both ways. Happy you keep on maintaining this repo. I’ve pushed a commit which addresses your comments.

LarsAlmgren commented:
@fsouza is there anything I need to fix in order for the CI to pass?

sig-daveka commented:
I'm eagerly awaiting having this functionality! Thanks for the effort!
